src/gkeClient.ts

```diff
@@ -226,38 +226,45 @@ export class ClusterClient {
   }
 
   /**
-   * Constructs a Connect Gateway endpoint string of form
-   * connectGatewayEndpoint/v1/projects/123/locations/l/memberships/m
+   * Constructs a Connect Gateway endpoint string from the given fleet
+   * membership name.
    *
-   * @param membershipName membership name.
+   * @param name membership name in the format projects/p/locations/l/memberships/m
    * @returns endpoint.
    */
-  async getConnectGWEndpoint(membershipName: string): Promise<string> {
-    const membershipRef = ClusterClient.parseMembershipName(membershipName);
-    const projectNum = await this.projectIDtoNum(membershipRef.projectID);
-    return `${this.connectGatewayHostPath}/projects/${projectNum}/locations/${membershipRef.location}/memberships/${membershipRef.membershipName}`;
-  }
-
-  /**
-   * Retrieves a Cluster.
-   *
-   * @param fqn Cluster name
-   * @returns a Cluster object.
-   */
-  async getCluster(clusterName: string): Promise<ClusterResponse> {
-    const url = `${this.defaultEndpoint}/${this.getResource(clusterName)}`;
+  async getConnectGWEndpoint(name: string): Promise<string> {
+    const membershipURL = `${this.hubEndpoint}/${name}`;
     const resp = (await this.auth.request({
-      url: url,
+      url: membershipURL,
       headers: {
         'User-Agent': userAgent,
       },
```

Comment:
Since we do a lookup, it might be good to call out any new permissions needed like `gkehub.memberships.get` in the connect gateway readme section.